indexing-service/src/main/java/org/apache/druid/indexing/overlord/hrtr/HttpRemoteTaskRunner.java

```diff
@@ -583,7 +583,7 @@ private void addWorker(final Worker worker)
           for (Map.Entry<String, HttpRemoteTaskRunnerWorkItem> e : tasks.entrySet()) {
             if (e.getValue().getState() == HttpRemoteTaskRunnerWorkItem.State.RUNNING) {
               Worker w = e.getValue().getWorker();
-              if (w != null && w.getHost().equals(worker.getHost())) {
+              if (w != null && w.getHost().equals(worker.getHost()) && e.getValue().getTask() != null) {
```

Comment:
I do not know this part of the system well, but I found a couple other places where this is called without a null check - `getPendingTaskPayloads` and `runTaskOnWorker`.

Could you annotate `HttpRemoteTaskRunner#getTask` with `@Nullable` so we can use intelliJ inspections to catch any uses of `getTask` without a null check.

Reply:
I see no risk in what was added, and your comment just delays the fix getting into the upstream. There is no mention or requirement for IntelliJ coding standards/annotations or whatever. If that bothers you that its not ideal in your IDE, why don't you raise a PR for IntelliJ (<your IDE here>) annotations and cleanups. Most likely the OP has already fixed in his fork and made this PR for upstream and the rest of us.  Now we have a stale PR (a valid fix) for something others need. Please get real, there are over 1300 open issues. Momentum would be nice, right?

Reply:
PR Authors can always agree or disagree with the code review. A UT was also requested that has not been added yet. While we want to fix the open issues, we do not want to compromise on the code quality. 

Also, IntelliJ inspections is a test job that we run for each PR. Suneet is referring to the inspections that run in the test job, not a local IDE.